tests/pipelines/test_pipelines_summarization.py

```diff
@@ -64,8 +65,15 @@ def run_pipeline_test(self, summarizer, _):
         if model.config.__class__.__name__ not in model_can_handle_longer_seq:
             # Switch Transformers, LED, T5, LongT5 can handle it.
             # Too long.
-            with self.assertRaises(Exception):
-                outputs = summarizer("This " * 1000)
+            # For TF models, if the weights are initialized in GPU context, we won't get expected index error from
+            # the embedding layer.
+            if not (
+                isinstance(model, TFPreTrainedModel)
+                and get_gpu_count() > 0
+                and len(summarizer.model.trainable_weights) > 0
```

Comment:
At the moment on `main`, this condition `len(summarizer.model.trainable_weights)` is `False`, as the model is not obtained from `from_pretrained` but by `model_class(config)` which won't create any TF weight at this point.

Reply:
So should the model be properly created with weights (guessing it needs a forward pass on dummy inputs)?

Reply:
@sgugger  On current `main`, there is no test failure around what I mentioned in this PR. We don't get the weights at line, and **they will be created inside pipeline forward** (which is being in **CPU context**, despite we have GPU), and we get the expected exception --> not test failure.

This PR is just to make the necessary changes in order my WIP PR could be merged without failure, where `len(summarizer.model.trainable_weights) > 0` will become `True` --> and we need to skip in this case.

Reply:
I still don't get why it's useful to support a model with no weights here. The fix is to make sure they have weights.